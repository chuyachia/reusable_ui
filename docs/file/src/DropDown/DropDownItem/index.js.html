<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <base data-ice="baseUrl" href="../../../../">
  <title data-ice="title">src/DropDown/DropDownItem/index.js | reusable-components-poc</title>
  <link type="text/css" rel="stylesheet" href="css/style.css">
  <link type="text/css" rel="stylesheet" href="css/prettify-tomorrow.css">
  <script src="script/prettify/prettify.js"></script>
  <script src="script/manual.js"></script>
</head>
<body class="layout-container" data-ice="rootContainer">

<header>
  <a href="./">Home</a>
  
  <a href="identifiers.html">Reference</a>
  <a href="source.html">Source</a>
  
  <div class="search-box">
  <span>
    <img src="./image/search.png">
    <span class="search-input-edge"></span><input class="search-input"><span class="search-input-edge"></span>
  </span>
    <ul class="search-result"></ul>
  </div>
</header>

<nav class="navigation" data-ice="nav"><div>
  <ul>
    
  <li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-useTheme">useTheme</a></span></span></li>
</ul>
</div>
</nav>

<div class="content" data-ice="content"><h1 data-ice="title">src/DropDown/DropDownItem/index.js</h1>
<pre class="source-code line-number raw-source-code"><code class="prettyprint linenums" data-ice="content">import styled from &quot;@emotion/styled&quot;;

import useTheme from &quot;../../useTheme&quot;;
import DropDownItem from &quot;./DropDownItem&quot;;

export default styled(DropDownItem)`
  list-style: none;
  cursor: pointer;
  padding: ${props =&gt; useTheme(props, &quot;basePadding&quot;)};
  font-family: ${props =&gt; useTheme(props, &quot;fontSans&quot;)};
  font-size: ${props =&gt; useTheme(props, &quot;baseFontSize&quot;)};
  font-weight: ${props =&gt; useTheme(props, &quot;fontNormal&quot;)};
  color: ${props =&gt; useTheme(props, &quot;baseTextColor&quot;)};
  line-height: ${props =&gt; useTheme(props, &quot;baseLineHeight&quot;)};
  &amp;:hover {
    background-color: ${props =&gt; {
      switch (props.variant) {
        case &quot;secondary&quot;:
          return useTheme(props, &quot;secondaryLight&quot;);
        case &quot;warning&quot;:
          return useTheme(props, &quot;warningLight&quot;);
        default:
          return useTheme(props, &quot;primaryLight&quot;);
      }
    }};
    font-weight: ${props =&gt; useTheme(props, &quot;fontSemibold&quot;)};
  }
`;
</code></pre>

</div>

<footer class="footer">
  Generated by <a href="https://esdoc.org">ESDoc<span data-ice="esdocVersion">(1.1.0)</span><img src="./image/esdoc-logo-mini-black.png"></a>
</footer>

<script src="script/search_index.js"></script>
<script src="script/search.js"></script>
<script src="script/pretty-print.js"></script>
<script src="script/inherited-summary.js"></script>
<script src="script/test-summary.js"></script>
<script src="script/inner-link.js"></script>
<script src="script/patch-for-local.js"></script>
</body>
</html>
