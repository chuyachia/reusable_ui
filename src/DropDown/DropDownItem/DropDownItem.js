import React from "react";
import PropTypes from "prop-types";
/**
 * General component description in JSDoc format. Markdown is *supported*.
 */
const DropDownItem = props => <li {...props} />;

DropDownItem.propTypes = {
  value: PropTypes.string,
  label: PropTypes.string,
  children: PropTypes.node,
  variant: PropTypes.string,
};

export default DropDownItem;
