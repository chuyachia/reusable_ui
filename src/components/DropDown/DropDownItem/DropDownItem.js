import React from "react";
import PropTypes from "prop-types";

const DropDownItem = props => <li {...props} />;

DropDownItem.propTypes = {
  value: PropTypes.string,
  children: PropTypes.node
};

export default DropDownItem;
