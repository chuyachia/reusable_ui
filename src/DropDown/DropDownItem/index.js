import styled from "@emotion/styled";

import useTheme from "../../useTheme";
import DropDownItem from "./DropDownItem";

export default styled(DropDownItem)`
  list-style: none;
  cursor: pointer;
  padding: ${props => useTheme(props, "basePadding")};
  font-family: ${props => useTheme(props, "fontSans")};
  font-size: ${props => useTheme(props, "baseFontSize")};
  font-weight: ${props => useTheme(props, "fontNormal")};
  color: ${props => {
    if (props.disabled) {
      return useTheme(props, "silver");
    }
    return useTheme(props, "baseTextColor");
  }};
  line-height: ${props => useTheme(props, "baseLineHeight")};
  &:hover,
  &.active {
    background-color: ${props => {
      if (props.variant) {
        return useTheme(props, props.variant + "Light");
      }
      return useTheme(props, "primaryLight");
    }};
  }
`;
