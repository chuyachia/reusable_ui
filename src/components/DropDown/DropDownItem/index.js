import styled from "@emotion/styled";

import DropDownItem from "./DropDownItem";

export default styled(DropDownItem)`
  list-style: none;
  cursor: pointer;
  padding: ${props => props.theme.basePadding};
  font-family: ${props => props.theme.fontSans};
  font-size: ${props => props.theme.baseFontSize};
  font-weight: ${props => props.theme.fontNormal};
  color: ${props => props.theme.baseTextColor};
  line-height: ${props => props.theme.baseLineHeight};
  &:hover {
    background-color: ${props => {
      switch (props.variant) {
        case "secondary":
          return props.theme.secondaryLight;
        case "warning":
          return props.theme.warningLight;
        default:
          return props.theme.primaryLight;
      }
    }};
    font-weight: ${props => props.theme.fontSemibold};
  }
`;
