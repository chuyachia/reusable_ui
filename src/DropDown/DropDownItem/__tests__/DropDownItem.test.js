import React from "react";

import DropDownItem from "../DropDownItem";

describe("DropDownItem", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<DropDownItem value="A" />);
  });
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should render the value property as a value <li>", () => {
    expect(wrapper.find("li").props().value).toBe("A");
  });
});
