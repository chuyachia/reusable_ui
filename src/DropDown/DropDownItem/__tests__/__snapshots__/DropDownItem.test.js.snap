// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownItem should render 1`] = `
<li
  value="A"
/>
`;
